File: routes/web.py
from flask import Blueprint, render_template, redirect, url_for, request
from models import FamilyMember

web_bp = Blueprint('web', __name__)

@web_bp.route('/')
def index():
    # If there are members, go to dashboard, otherwise redirect to profiles to create one
    first_member = FamilyMember.query.first()
    if first_member:
        return redirect(url_for('web.dashboard', member_id=first_member.id))
    return redirect(url_for('web.profiles'))

@web_bp.route('/dashboard')
def dashboard():
    member_id = request.args.get('member_id', type=int)
    members = FamilyMember.query.all()
    
    selected_member = None
    if member_id:
        selected_member = FamilyMember.query.get(member_id)
    elif members:
        selected_member = members[0]
        
    return render_template(
        'dashboard.html',
        members=members,
        selected_member=selected_member
    )

@web_bp.route('/profiles')
def profiles():
    members = FamilyMember.query.all()
    return render_template('profiles.html', members=members)

@web_bp.route('/chat')
def chat():
    member_id = request.args.get('member_id', type=int)
    members = FamilyMember.query.all()
    
    selected_member = None
    if member_id:
        selected_member = FamilyMember.query.get(member_id)
    elif members:
        selected_member = members[0]
        
    return render_template(
        'chat.html',
        members=members,
        selected_member=selected_member
    )

@web_bp.route('/meal-planner')
def meal_planner():
    member_id = request.args.get('member_id', type=int)
    members = FamilyMember.query.all()
    
    selected_member = None
    if member_id:
        selected_member = FamilyMember.query.get(member_id)
    elif members:
        selected_member = members[0]
        
    # Get the latest meal plan if any exists
    latest_plan = None
    if selected_member:
        from models.meal_plan import MealPlan
        latest_plan = MealPlan.query.filter_by(member_id=selected_member.id).order_by(MealPlan.created_at.desc()).first()

    return render_template(
        'meal_planner.html',
        members=members,
        selected_member=selected_member,
        latest_plan=latest_plan
    )

@web_bp.route('/bmi')
def bmi_calculator():
    # Standalone BMI page that doesn't strictly require saving, but can link to profile creation
    return render_template('bmi.html')

File: routes/web.js
const express = require('express');
const router = express.Router();
const { FamilyMember, MealPlan } = require('../models');

router.get('/', async (req, res) => {
  try {
    const firstMember = await FamilyMember.findOne();
    if (firstMember) {
      return res.redirect(`/dashboard?member_id=${firstMember.id}`);
    }
    return res.redirect('/profiles');
  } catch (err) {
    console.error(err);
    res.status(500).send('Internal Server Error');
  }
});

router.get('/dashboard', async (req, res) => {
  try {
    const memberId = req.query.member_id ? parseInt(req.query.member_id) : null;
    const members = await FamilyMember.findAll();

    let selectedMember = null;
    if (memberId) {
      selectedMember = await FamilyMember.findByPk(memberId);
    } else if (members.length > 0) {
      selectedMember = members[0];
    }

    res.render('dashboard', {
      title: 'Nutrition Dashboard - Arogya AI',
      members,
      selected_member: selectedMember,
      path: '/dashboard'
    });
  } catch (err) {
    console.error(err);
    res.status(500).send('Internal Server Error');
  }
});

router.get('/profiles', async (req, res) => {
  try {
    const members = await FamilyMember.findAll();
    res.render('profiles', {
      title: 'Family Profiles - Arogya AI',
      members,
      path: '/profiles'
    });
  } catch (err) {
    console.error(err);
    res.status(500).send('Internal Server Error');
  }
});

router.get('/chat', async (req, res) => {
  try {
    const memberId = req.query.member_id ? parseInt(req.query.member_id) : null;
    const members = await FamilyMember.findAll();

    let selectedMember = null;
    if (memberId) {
      selectedMember = await FamilyMember.findByPk(memberId);
    } else if (members.length > 0) {
      selectedMember = members[0];
    }

    res.render('chat', {
      title: 'Chat with Arogya AI - Personal Nutrition Assistant',
      members,
      selected_member: selectedMember,
      path: '/chat'
    });
  } catch (err) {
    console.error(err);
    res.status(500).send('Internal Server Error');
  }
});

router.get('/meal-planner', async (req, res) => {
  try {
    const memberId = req.query.member_id ? parseInt(req.query.member_id) : null;
    const members = await FamilyMember.findAll();

    let selectedMember = null;
    if (memberId) {
      selectedMember = await FamilyMember.findByPk(memberId);
    } else if (members.length > 0) {
      selectedMember = members[0];
    }

    let latestPlan = null;
    if (selectedMember) {
      latestPlan = await MealPlan.findOne({
        where: { member_id: selectedMember.id },
        order: [['created_at', 'DESC']]
      });
    }

    res.render('meal_planner', {
      title: 'Meal Planner - Arogya AI',
      members,
      selected_member: selectedMember,
      latest_plan: latestPlan,
      path: '/meal-planner'
    });
  } catch (err) {
    console.error(err);
    res.status(500).send('Internal Server Error');
  }
});

router.get('/bmi', (req, res) => {
  res.render('bmi', {
    title: 'BMI Calculator - Arogya AI',
    path: '/bmi'
  });
});

module.exports = router;
